setup.py

```diff
@@ -200,9 +203,38 @@ def build_extensions(self):
         return super().build_extensions()
 
 
-cmdclass = versioneer.get_cmdclass()
+def update_matplotlibrc(path):
+    # Update the matplotlibrc file if packagers want to change the default
+    # backend.
+    template_lines = path.read_text().splitlines(True)
+    backend_line_idx, = [  # Also asserts that there is a single such line.
+        idx for idx, line in enumerate(template_lines)
+        if line.startswith("#backend:")]
+    template_lines[backend_line_idx] = (
+        "#backend: {}".format(setupext.options["backend"])
+        if setupext.options["backend"]
+        else "#backend:")
```

Comment:
The template currently say

```
#backend: Agg
```

so this is not a no-op in any case.  I think we should change the checked-in matplotlibrc file to match this.

It took me a bit to sort out again, but the backend default is explicitly not inferred from this file (as it defaults to a sentinel object).